spec/features/json_spec.rb

```diff
@@ -82,18 +80,17 @@ class JSONTest < ActionDispatch::IntegrationTest
           ],
         },
       }
-      actual = JSON.parse(@response.body)
+      actual = JSON.parse(page.body)
       expected.each do |nation, expected_bank_holidays|
         actual_bank_holidays = actual.fetch(nation)
-        assert((expected_bank_holidays["events"] - actual_bank_holidays["events"]).empty?)
-        assert_equal expected_bank_holidays["division"], actual_bank_holidays["division"]
+        expect(expected_bank_holidays["events"] - actual_bank_holidays["events"]).to be_empty
+        expect(actual_bank_holidays["division"]).to eq(expected_bank_holidays["division"])
       end
     end
 
-    should "have redirect for old 'ni' division" do
-      get "/bank-holidays/ni.json"
-      assert_equal 301, response.status
```

Comment:
Do we lose something by not asserting this is a 301? For example, it could now be a 302 and the test would still pass.

Reply:
The problem is that capybara follows the redirect, so it sees a status code of 200.

I could do something like the `do_not_follow_redirect` solution on https://exceptionshub.com/how-to-test-for-a-redirect-with-rspec-and-capybara.html if you think that would be best?

Reply:
Shame, [Capybara doesn't offer a way of asserting 301 responses](https://github.com/teamcapybara/capybara/issues/1149), and the `do_not_follow_redirect` solution looks a bit gross! Yes let's stick with what you have 👍